server/parser/src/main/java/org/eclipse/lsp/cobol/core/hw/CobolParser.java

```diff
@@ -321,7 +330,32 @@ private void or(String... expectedTokens) {
   }
 
   boolean sameLexeme(String lexeme1, String lexeme2) {
-    return lexeme1 != null && lexeme2 != null && Objects.equals(lexeme1.toUpperCase(), lexeme2.toUpperCase());
+    if (lexeme1 == null || lexeme2 == null) {
+      return false;
+    }
+    Double apply = sim.apply(lexeme1.toUpperCase(), lexeme2.toUpperCase());
+    if (apply == 1) {
+        return true;
+    }
+    if (apply > SIMILARITY_THRESHOLD){
+      diagnostics.add(getSimilarKeywordPassedSyntaxError(lexeme1, lexeme2));
+      return true;
+    }
+    return false;
+  }
+
+  private SyntaxError getSimilarKeywordPassedSyntaxError(String lexeme1, String lexeme2) {
```

Comment:
Not sure , if about this change.
I will rework to get token locality for diagnostics